src/main/ruby/truffleruby/core/integer.rb

```diff
@@ -122,8 +122,9 @@ def nobits?(mask)
 
   def pow(e, m=undefined)
     return self ** e if Primitive.undefined?(m)
+    raise TypeError, '2nd argument not allowed unless a 1st argument is integer' unless Primitive.object_kind_of?(e, Integer)
     raise TypeError, '2nd argument not allowed unless all arguments are integers' unless Primitive.object_kind_of?(m, Integer)
-    (self ** e) % m
+    Primitive.mod_pow(self, e, m)
```

Comment:
Could you refactor this as a `if/else`? I think that would be easier to read than the early return now that there is more logic in the 2-arguments case.